docs/IR.md

```diff
@@ -109,6 +108,20 @@ considered Constant even if only ever read by other instructions. Thus, the
 visibility of the Variable is also annotated on its generated WeightVar, and all
 Public WeightVars are kept Mutable.
 
+### Placeholders
+
+Placeholders are symbolic nodes that are not backed by a concrete tensor during
+the compilation of the program. Inputs and outputs of Glow programs should be
+modeled using Placeholder nodes. Concrete tensors are attached to placeholder
```

Comment:
@rdzhabarov At the moment it happens at the compilation phase. @opti-mix mentioned that it could be useful to bind the tensors at the run() phase. There are tradeoffs here. It should be relatively straightforward to change this later on.